mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -539,7 +540,14 @@ def create_run(self, experiment_id, user_id, start_time, tags):
                 lifecycle_stage=LifecycleStage.ACTIVE,
             )
 
-            run.tags = [SqlTag(key=tag.key, value=tag.value) for tag in tags] if tags else []
+            if MLFLOW_RUN_NAME not in [tag.key for tag in tags]:
+                run_name_tag = RunTag(MLFLOW_RUN_NAME, _generate_random_name())
+                if tags:
```

Comment:
if `tags` were `None`, then line 543 would fail. I think we can get rid of the if / else and always call `append()`. If there's any possibility of tags being `None`, we should validate it prior to line 543.

Reply:
changed!